src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -2281,16 +2281,427 @@ static void F(ref T x)
+                //         GetRef(ref s1) = TRefStruct.New(ref value); // 4
+                Diagnostic(ErrorCode.ERR_RefReturnParameter, "value").WithArguments("value").WithLocation(22, 45)
+                );
+        }
+
+        // This is a clone of ReturnRefToRefStruct_ValEscape_04 from RefFieldTests.cs
+        [Fact]
+        public void ReturnRefToRefStruct_ValEscape_04()
+        {
+            // test that the appropriate filtering of escape-values is occurring when the RTRS expression is on the RHS of an an assignment.
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+
+                class Repro<TRefStruct> where TRefStruct : IRefStruct<TRefStruct>, allows ref struct
+                {
+                    private static void M1(ref TRefStruct s1, int value)
+                    {
+                        // 's2' only contributes STE, not RSTE, to the STE of 'RefMethod()' invocation.
+                        // STE is equal to RSTE for 's2', so it doesn't matter.
+                        var s2 = TRefStruct.New(ref value);
+                        s1 = s2.RefMethod(); // 1
+                    }
+                    
+                    private static void M2(ref TRefStruct s1, ref TRefStruct s2)
+                    {
+                        // 's2' only contributes STE, not RSTE, to the STE of 'RefMethod()' invocation.
+                        // RSTE of `s2` is narrower than STE of 's1', but STE of 's2' equals STE of 's1', so we expect no error here.
+                        s1 = s2.RefMethod();
+                    }
+                }
+                
+                interface IRefStruct<TRefStruct> where TRefStruct : IRefStruct<TRefStruct>, allows ref struct
+                {
+                    abstract static TRefStruct New(ref int i);
+                    [UnscopedRef] public ref TRefStruct RefMethod();
+                }
+                """;
+            var comp = CreateCompilation(source, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (10,14): error CS8352: Cannot use variable 's2' in this context because it may expose referenced variables outside of their declaration scope
+                //         s1 = s2.RefMethod(); // 1
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "s2").WithArguments("s2").WithLocation(10, 14));
+        }
+
+        // This is a clone of LocalScope_DeclarationExpression_06 from RefEscapingTests.cs
+        [Fact]
+        public void LocalScope_DeclarationExpression_06()
+        {
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+
+                interface IRS<RS> where RS : IRS<RS>, allows ref struct
+                {
+                    [UnscopedRef]
+                    void M0(out RS rs2);
+
+                    RS M1()
+                    {
+                        // RSTE of `this` is CurrentMethod
```

Comment:
>this [](http://example.com/codeflow?start=36&length=4)

`this` is an interface, not a type parameter with `allows ref struct`.

Reply:
> `this` is an interface, not a type parameter with `allows ref struct`.

And the point you are trying to make is?

Reply:
The test didn't seem particularly interesting to me because `this` is an interface and not by-ref-like. But the behavior (no errors reported) looks correct.

Reply:
The behavior is different by comparison to the cloned test, but correct. That feels good enough to keep the test.